akka-diagnostics/src/test/scala/akka/diagnostics/StarvationDetectorSpec.scala

```diff
@@ -184,6 +174,25 @@ class StarvationDetectorSpec extends AkkaSpec(s"""
       Try(f.get(100, TimeUnit.MILLISECONDS))
     })
 
+  //As per https://openjdk.org/jeps/353 in Java 13: "The new implementation, NioSocketImpl, is a drop-in replacement for PlainSocketImpl."
+  // "Socket operations using timeouts (connect, accept, read) are implemented by changing the socket to non-blocking mode and polling the socket."
```

Comment:
The blocking mentioned there is about how it interacts with the socket, not the calling thread. So the two examples below are still blocking the calling thread (until possibly in the future with loom where it could be a green thread - from the JEP "In the context of a future world of fibers that park instead of blocking threads in native methods, ...")